sklearn/linear_model/tests/test_logistic.py

```diff
@@ -1136,3 +1136,25 @@ def test_saga_vs_liblinear():
                 liblinear.fit(X, y)
                 # Convergence for alpha=1e-3 is very slow
                 assert_array_almost_equal(saga.coef_, liblinear.coef_, 3)
+
+
+def test_dtype_match():
+    # Test that np.float32 input data is not cast to np.float64 when possible
+
+    X_32 = np.array(X).astype(np.float32)
+    y_32 = np.array(Y1).astype(np.float32)
+    X_64 = np.array(X).astype(np.float64)
+    y_64 = np.array(Y1).astype(np.float64)
+
+    for solver in ['newton-cg']:
```

Comment:
Not all the solvers are safe with cast to float32. The idea is to have `X_64` and `X_32` add the not safe solver, brake the test and track the failure.